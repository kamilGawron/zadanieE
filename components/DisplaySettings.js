import React,{useState} from 'react'

function DisplaySettings(props){
    return(
        <div className="settings">
            <span>Samochody ({props.cars.length})</span>
            <label className="switch">
                <input id="carsToggler" type="checkbox" onChange={props.carsToggler} defaultChecked/>
                <span className="slider round"></span>
            </label>
            <span>Parkingi ({props.parkings.length})</span>
            <label className="switch">
                <input id="parkingsToggler" type="checkbox" onChange={props.parkingsToggler} defaultChecked/>
                <span className="slider round"></span>
            </label>
            <span>POI ({props.pois.length})</span>
            <label className="switch">
                <input id="poisToggler" type="checkbox" onChange={props.poisToggler} defaultChecked/>
                <span className="slider round"></span>
            </label>
            <label>Minimalny procentowy poziom baterii samochodu
                <input name="minBatteryLevel" onChange={props.inputChange} min="0" max="100" id="minBatteryInput" type="range" value={props.minBatteryLevel}/>
                <input name="minBatteryLevel" onChange={props.inputChange} min="0" max="100" id="minBatteryInput" type="text" value={props.minBatteryLevel}/>
            </label>
            <label>Minimalny zasięg kilometrów
                <input name="minRange" onChange={props.inputChange} min="0" id="minRangeInput" type="number" value={props.minRange}/>
                <input name="minRange" onChange={props.inputChange} min="0" max={props.maxRange} id="minRangeInput" type="range" value={props.minRange}/>
            </label>
            {
             props.minBatteryLevel!=0||
             props.minRange!=0?
                 props.filterCarLen==props.cars.length?
                 <span>Wszystkie samochody spełniają określone kryteria</span>:
                 <span>znaleziono: {props.filterCarLen}</span> 
             : ""
            }
        </div>
       
    )
}

export default DisplaySettings