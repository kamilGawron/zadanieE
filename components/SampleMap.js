
import React, { Component } from 'react'
import { Map, TileLayer, Marker, Popup } from 'react-leaflet'
import MarkerClusterGroup from 'react-leaflet-markercluster';
import L from 'leaflet'

var carIcon = L.icon({
    iconUrl: '../img/car.png',

    iconSize:     [32, 32], // size of the icon
    iconAnchor:   [16, 16], // point of the icon which will correspond to marker's location
    popupAnchor:  [0, -24] // point from which the popup should open relative to the iconAnchor
});
var parkingIcon = L.icon({
    iconUrl: '../img/parking.png',

    iconSize:     [32, 32], // size of the icon
    iconAnchor:   [16, 16], // point of the icon which will correspond to marker's location
    popupAnchor:  [0, -24] // point from which the popup should open relative to the iconAnchor
});
var poiIcon = L.icon({
    iconUrl: '../img/poi.png',

    iconSize:     [32, 32], // size of the icon
    iconAnchor:   [16, 16], // point of the icon which will correspond to marker's location
    popupAnchor:  [0, -24] // point from which the popup should open relative to the iconAnchor
});


export default class SampleMap extends Component {
    constructor(props){
        super(props);
        this.state={
            lat: 51.919438,
            lng: 19.145136,
            zoom: 7,
            cars:[],
            carsLoaded:false,
            parkings:[],
            parkingsLoaded:false,
            pois:[],
            poisLoaded:false,
        }
}
    componentDidMount(){
        
        fetch("https://dev.vozilla.pl/api-client-portal/map?objectType=VEHICLE")
        .then(response=>response.json())
        .then(function(data){
            console.log("Cars",data.objects);
            return data
        })
        .then((data)=>{
            this.setState({cars:data.objects,carsLoaded:true})
        })
        fetch("https://dev.vozilla.pl/api-client-portal/map?objectType=PARKING")
            .then(response=>response.json())
            .then(function(data){
                console.log("parking",data.objects);
                return data
            })
            .then((data)=>{
                this.setState({parkings:data.objects,parkingsLoaded:true})
        })
        fetch("https://dev.vozilla.pl/api-client-portal/map?objectType=POI")
            .then(response=>response.json())
            .then(data=>{
                this.props.setCarsCounter();
                console.log("POI",data.objects);
                return data
            })
            .then((data)=>{
                this.setState({pois:data.objects,poisLoaded:true})
            })
        }
   componentDidUpdate(){
       
   }
   
render() {
    let cars,parkings,pois;
    var self = this;
    if(this.state.carsLoaded){
        let tmpCars = this.state.cars.filter(function(elem){
            return elem.batteryLevelPct>self.props.minBatteryLevel
        })
        cars = this.props.showCars?  tmpCars.map(function(elem,index){
            return(
                <Marker 
                    key={index} 
                    position={[elem.location.latitude,elem.location.longitude]}
                    icon ={carIcon}
                    >
                    <Popup>
                        <React.Fragment>{elem.name}</React.Fragment>
                    </Popup>
                </Marker>
            )
        }) : []
    }
    if(this.state.parkingsLoaded){
        parkings = this.props.showParkings? this.state.parkings.map(function(elem,index){
            return(
                <Marker 
                    key={index} 
                    position={[elem.location.latitude,elem.location.longitude]}
                    icon ={parkingIcon}
                    >
                    <Popup>
                        <React.Fragment>{elem.name}</React.Fragment>
                    </Popup>
                </Marker>
            )
        }) : []
    }
    if(this.state.poisLoaded){
        pois = this.props.showPois? this.state.pois.map(function(elem,index){
            return(
                <Marker 
                    key={index} 
                    position={[elem.location.latitude,elem.location.longitude]}
                    icon ={poiIcon}
                    >
                    <Popup>
                        <React.Fragment>{elem.name}</React.Fragment>
                    </Popup>
                </Marker>
            )
        }) :[]
    }
    
    
    return (
        this.state.carsLoaded?
        <Map center={[this.state.lat,this.state.lng]} zoom={this.state.zoom}>
            <TileLayer
                attribution='&amp;copy <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
                url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
            />
           
            <MarkerClusterGroup>
                {cars}
            </MarkerClusterGroup>
            <MarkerClusterGroup>
                {parkings}
            </MarkerClusterGroup>
            <MarkerClusterGroup>
                {pois}
            </MarkerClusterGroup>
        </Map>
        :
        <div>loading</div>
    )
}
}