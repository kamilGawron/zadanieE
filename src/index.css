:root{
    font-size: 10px;
}
body{
    padding: 0;
    margin: 0;
}
.loading{
    height: 100vh;
    width: 100vw;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    background-color: rgba(0,0,0,.9);
    font-size: 2rem;
    color: #ffffff;
    text-transform: uppercase;
}
.loading .spiner{
    margin-top: 30px;
    height: 80px;
    width: 80px;
    border-radius: 50%;
    border: 5px solid rgba(0,0,0,0);
    border-bottom: 5px solid #2196F3;
    -webkit-animation: spiner .7s ease-in-out;
    animation: spiner .8s ease-in-out infinite;
}
@-webkit-keyframes spiner{
    0%{
        -webkit-transform: rotate(0);
        transform: rotate(0);
    }
    100%{
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg)
    }
}
@keyframes spiner{
    0%{
        -webkit-transform: rotate(0);
        transform: rotate(0);
    }
    100%{
        -webkit-transform: rotate(360deg);
        transform: rotate(360deg)
    }
}
.settings{
    z-index: 1000;
    position: absolute;
    top: 0;
    left: 0;
    height: 100px;
    width: 100%;
    background-color: rgba(0,0,0,.9);
    border-bottom: 2px solid #2196F3;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
}
.settings .general{
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex; 
    -webkit-box-orient: vertical; 
    -webkit-box-direction: normal; 
    -ms-flex-direction: column; 
    flex-direction: column;
    -webkit-box-pack: center; 
    -ms-flex-pack: center; 
    justify-content: center;
    color: #ffffff;
    font-size: 1.4rem;
}
.settings .general>h6{
    margin: 5px;
    text-align: center;
    font-size: 1.6rem;
}
.settings .general>div{
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
}
.settings .general>div>div{
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    margin: 10px;
    width: 100px;
}
.settings .general div:last-of-type{
    margin-left: 50px;
    
}
.settings .general div .filters-icon{
    font-size: 2.4rem;
    cursor: pointer;
    color: #ffffff;
    -webkit-transition: all .7s ease-in-out;
    -webkit-transition: all .3s ease-in-out;
    transition: all .3s ease-in-out;
}
.settings .general div .filters-icon:hover{
    color: #2196F3;
}
.settings #filters{
    position: absolute;
    top:100px;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    width: 60%;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    background-color: rgba(0,0,0,.9);
    padding: 10px;
    padding-left: 50px;
    border-radius: 20px;
    color: #ffffff;
    font-size: 1.4rem;

    
}
.settings .specific-settings .close{
    position: absolute;
    right: 20px;
    top:20px;
    font-size: 2.5rem;
    font-weight: bold;
    cursor: pointer;
    color: #ffffff;
    -webkit-transition: all .7s ease-in-out;
    -webkit-transition: all .3s ease-in-out;
    transition: all .3s ease-in-out;
}
.settings .specific-settings .close:hover{
    color: #2196F3;
}
.settings .specific-settings input::-webkit-outer-spin-button,
.settings .specific-settings input::-webkit-inner-spin-button {
    /* display: none; <- Crashes Chrome on hover */
    -webkit-appearance: none;
    margin: 0; 
}

.settings .specific-settings input[type=number]{
    width: 40px;
    border: none;
    background-color: transparent;
    color: #ffffff;
    border-bottom: 1px solid #2196F3;
    -moz-appearance:textfield;
    text-align: center;
    outline: none;
}
.settings .specific-settings input[type=range]::-webkit-slider-thumb,
.settings .specific-settings input[type=range]::-moz-range-thumb,
.settings .specific-settings input[type=range]::-ms-thumb{
   color: red;
}

.settings .specific-settings .cars,
.settings .specific-settings .parkings{
    margin: 0;
    width: 50%;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
}
.settings .specific-settings .cars>div,
.settings .specific-settings .parkings>div{
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
}
.settings .specific-settings .cars>div .switch{
    top: 0;
    left: 10px;
}
.settings .specific-settings .cars h6,
.settings .specific-settings .parkings h6{
    margin: 0 0 10px 0;
    font-size: 1.6rem;
}
.settings .specific-settings div>span{
    margin: 10px 0;
    font-weight: bold;
    color: #2196F3;
    font-size: 1.4rem;
}
.settings .specific-settings .clear-input{
    color: #ffffff;
    cursor: pointer;
    height: 20px;
    width: 20px;
    -webkit-transition: all .3s ease-in-out;
    transition: all .3s ease-in-out;
}
.settings .specific-settings .clear-input:hover{
    color: #2196F3;
}
.switch {
    position: relative;
    display: inline-block;
    width: 46px;
    height: 25px;
    top:5px;
}

.switch input {
    opacity: 0;
    width: 0;
    height: 0;
}

/* The slider */
.slider {
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: #ccc;
    -webkit-transition: .4s;
    transition: .4s;
}

.slider:before {
    position: absolute;
    content: "";
    height: 16px;
    width: 16px;
    left: 4px;
    bottom: 4px;
    background-color: white;
    -webkit-transition: .4s;
    transition: .4s;
}

input:checked + .slider {
    background-color: #2196F3;
}

input:focus + .slider {
    -webkit-box-shadow: 0 0 1px #2196F3;
    box-shadow: 0 0 1px #2196F3;
}

input:checked + .slider:before {
    -webkit-transform: translateX(22px);
    transform: translateX(22px);
}

/* Rounded sliders */
.slider.round {
    border-radius: 34px;
}

.slider.round:before {
    border-radius: 50%;
}

.leaflet-container{
    height: 100vh;
    width: 100vw;
    margin: 0;
}
.leaflet-control{
    top:100px;
}
.leaflet-popup-content-wrapper{
    background-color: rgba(0,0,0,.9);
    color: #ffffff;
}
.leaflet-popup-content h3{
    margin: 5px 0;
    text-align: center;
    color: #2196F3;
}
.leaflet-popup-content .car>div:first-of-type{
    text-align: center;
    font-weight: bold;
    margin:0 0 5px 0;
}
.leaflet-popup-content .content .title{
    font-weight: bold;
    margin-right: 5px;
}
.leaflet-popup-content .poi>div:first-of-type{
    text-align: center;
    margin-bottom: 5px;
    
}
.leaflet-popup-tip{
    background-color: rgba(0,0,0,.9);
}
.leaflet-container a.leaflet-popup-close-button{
    color: #ffffff;
    -webkit-transition: all .3s ease-in-out;
    transition: all .3s ease-in-out;
}
.leaflet-container a.leaflet-popup-close-button:hover{
    color:  #2196F3;
}
@media all and (max-width:768px){
    .settings #filters{
        width: 100%;
        border-radius: 0;
    }
    .settings .specific-settings input[type=range]{
        width: 120px;
    }
    .settings .general>div{
        padding: 0;
        -webkit-box-pack: start;
        -ms-flex-pack: start;
    }
    .settings .general>div>div{
        width: 90px;
        font-size: 1.1rem;
        margin: 0px;
    }
    .settings .general div:last-of-type{
        margin-left: 0px;
        
    }
    .settings #filters{
        -webkit-box-orient: vertical;
        -webkit-box-direction: normal;
        -ms-flex-direction: column;
        flex-direction: column;
    }
    .settings .specific-settings .cars,
    .settings .specific-settings .parkings{
        width: 100%;
    }
    .settings .specific-settings .parkings{
        margin-top: 10px;
    }
    
}