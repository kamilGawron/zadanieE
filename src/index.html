<!DOCTYPE html>
<html lang="">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title></title>
    <link rel="stylesheet" href="https://unpkg.com/leaflet@1.0.1/dist/leaflet.css" />

</head>

<body>
   <div id="root"></div>
    <script src=""></script>
    
</body>
</html>
