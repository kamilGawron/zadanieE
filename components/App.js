import React from 'react'
import SampleMap from './SampleMap'
import DisplaySettings from './DisplaySettings'
import "../node_modules/leaflet/dist/leaflet.css"
import "../node_modules/react-leaflet-markercluster/dist/styles.min.css"

class App extends React.Component{
    constructor(){
        super();
        this.state={
            cars:[],
            carsLoaded:false,
            parkings:[],
            parkingsLoaded:false,
            pois:[],
            poisLoaded:false,
            showCars:true,
            showParkings:true,
            showPois:true,
            minBatteryLevel:0,
            minRange:0,
            maxRange:0,
            minSpaces:0,
            maxSpaces:0
        }
        this.carsToggler = this.carsToggler.bind(this)
        this.parkingsToggler = this.parkingsToggler.bind(this)
        this.poisToggler = this.poisToggler.bind(this)
        this.inputChange = this.inputChange.bind(this)
    }
    componentDidMount(){
        let tmpMaxRange=0,tmpMaxSpaces=0;
        fetch("https://dev.vozilla.pl/api-client-portal/map?objectType=VEHICLE")
            .then(response=>response.json())
            .then((data)=>{
                tmpMaxRange = data.objects[0].rangeKm;
                for (let x in data.objects){
                    if(data.objects[x].rangeKm>tmpMaxRange){
                        tmpMaxRange = data.objects[x].rangeKm;
                    }
                }
                this.setState({cars:data.objects,carsLoaded:true,maxRange:tmpMaxRange})
            })
        fetch("https://dev.vozilla.pl/api-client-portal/map?objectType=PARKING")
            .then(response=>response.json())
            .then((data)=>{
                tmpMaxSpaces = data.objects[0].availableSpacesCount;
                for (let x in data.objects){
                    if(data.objects[x].availableSpacesCount>tmpMaxSpaces){
                        tmpMaxSpaces = data.objects[x].availableSpacesCount;
                    }
                }
                this.setState({parkings:data.objects,parkingsLoaded:true,maxSpaces:tmpMaxSpaces})
            })
        fetch("https://dev.vozilla.pl/api-client-portal/map?objectType=POI")
            .then(response=>response.json())
            .then((data)=>{
                this.setState({pois:data.objects,poisLoaded:true})
            })
        }
    
    carsToggler(){
        this.setState(prev=>{
            return{
                showCars:!prev.showCars
            }
        })
    }
    parkingsToggler(){
        this.setState(prev=>{
            return{
                showParkings:!prev.showParkings
            }
        })
    }
    poisToggler(){
        this.setState(prev=>{
            return{
                showPois:!prev.showPois
            }
        })
    }
    inputChange(e){
        this.setState({[e.target.name]:e.target.value})
    }
    render(){
        let tmpCars,tmpParkings;
        if(this.state.carsLoaded){
            tmpCars = this.state.cars.filter(elem=>{
                return elem.batteryLevelPct>=this.state.minBatteryLevel&&elem.rangeKm>=this.state.minRange
            })
        }
        if(this.state.parkingsLoaded){
            tmpParkings = this.state.parkings.filter(elem=>{
                return elem.availableSpacesCount>=this.state.minSpaces
            })
        }
        return(
            this.state.carsLoaded&&this.state.parkingsLoaded&&this.state.poisLoaded?
                <div>
                   <DisplaySettings
                      {...this.state}
                       carsToggler={this.carsToggler}
                       parkingsToggler={this.parkingsToggler}
                       poisToggler={this.poisToggler}
                       inputChange={this.inputChange}
                       filterCarLen={tmpCars.length}
                       filterParkingLen={tmpParkings.length}
                   />
                    <SampleMap
                        {...this.state}
                        cars={tmpCars}
                        parkings={tmpParkings}
                    />
                </div>
                : 
                <div className="loading">
                   <div>
                       loading...
                   </div>
                   <div className="spiner"></div>
                </div>
        )
    }
}
export default App