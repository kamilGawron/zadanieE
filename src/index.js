import React from 'react'
import ReactDOM from 'react-dom'
import App from '../components/App.js'
require('../style/index.css');
require('../style/media.css');
require('../style/spinner.css');
require('../style/map.css');
require('../style/settings.css');


ReactDOM.render(<App />,document.getElementById('root'))