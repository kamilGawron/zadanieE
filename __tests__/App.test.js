import React from 'react';
import { shallow } from 'enzyme';
import App from '../components/App';
import DisplaySettings from '../components/DisplaySettings';
import SampleMap from '../components/SampleMap';
import '../src/setupTest'
import renderer from 'react-test-renderer'
describe('test <App /> when load state is false', () => {
    const wrapper = shallow(<App/>);
    it('matches the snapshot', () => {
        const tree=renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('does not renders a <DisplaySettings/>', () => {
        expect(wrapper.find(DisplaySettings)).toHaveLength(0);
    });
    it('does not renders a <SampleMap/>', () => {
        expect(wrapper.find(SampleMap)).toHaveLength(0);
    });
});
describe('test <App /> when load state is true', () => {
    const wrapper = shallow(<App/>);
    wrapper.setState({
        carsLoaded:true,
        parkingsLoaded:true,
        poisLoaded:true,
        fetchErr:false,
    })
    it('renders a <DisplaySettings/>', () => {
        expect(wrapper.find(DisplaySettings)).toHaveLength(1);
    });
    it('renders a <SampleMap/>', () => {
        expect(wrapper.find(SampleMap)).toHaveLength(1);
    });
});