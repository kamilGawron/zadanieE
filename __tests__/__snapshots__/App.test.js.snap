// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <App /> when load state is false matches the snapshot 1`] = `
<div
  className="loading"
>
  <div>
    Å‚adowanie
  </div>
  <div
    className="spinner"
  />
</div>
`;
